README.md

```diff
@@ -1,154 +1,193 @@
+This makes verification trivial: hash the header to derive the siphash key,
+then compute the 42x2 edge endpoints with 84 siphashes, check that each endpoint occurs twice,
+and that you come back to the starting point only after traversing 42 edges.
 
-From this point of view, Cuckoo Cycle is a very simple PoW,
-requiring hardly any code, time, or memory to verify.
-
-Finding a 42-cycle, on the other hand, is far from trivial,
+While trivially verifiable, finding a 42-cycle, on the other hand, is far from trivial,
 requiring considerable resources, and some luck
-(for a given header, the odds of its graph having a L-cycle are about 1 in L).
+(the odds of a random cuckoo graph having an L-cycle are approximately 1 in L).
 
+Lean mining
+-----------
 The memory efficient miner uses 3 bits per edge and is bottlenecked by
-accessing random 2-bit counters, making it memory latency bound.  The roughly
-4x faster latency avoiding miner, a rewrite from xenoncat's bounty winning solver,
-uses 33 bits per edge and is bottlenecked by bucket sorting. making it memory bandwidth bound.
+accessing random 2-bit counters, making it memory latency bound.
+The core of this miner, where over 99% of time is spent, is also [relatively simple](doc/leancore).
+
+Mean mining
+-----------
+The roughly 4x faster latency avoiding miner uses 33 bits per edge
+and is bottlenecked by bucket sorting, making it memory bandwidth bound.
 
-ASICs
+Dynamic Sizing
 --------------
-The most cost effective Cuckoo Cycle mining hardware could consist of a relatively cheap and tiny ASIC
-containing a few dozen simple cores and memory controllers, coupled with 3D-stacked DRAM using
-the <a href="https://en.wikipedia.org/wiki/High_Bandwidth_Memory">High Bandwidth Memory</a> interface.
-The ASIC, running the faster bandwidth bound solver, wouldn't be very compute intensive.
-Since it only needs to keep the memory bandwidth saturated, its optimization would soon reach a point of
-diminishing returns. The hardware and energy costs will be dominated by the memory chips,
-which the commodity DRAM market is already constantly optimizing.
-Although running the memory efficient latency bound solver would allow for processing many graph
-instances simultaneously, the induced latencies would multiply and overall throughput remain lower.
-Unless, that is, a new kind of RAM were developed that is much more cost efficient for random accesses.
+Instead of fixing N, a proof-of-work system could allow miners to work on any graph size they like,
+above a certain minimum. Cycles in larger graphs are more valued as they take more effort to find.
+We propose to scale the difficulty for a graph on 2^N nodes by 2^N * (N-1),
+which is the number of bits of siphash output that define the graph.
+
+Upgrade by Soft Fork
+--------------------
+Dynamic Sizing allows for increasing the memory required for mining simply by raising the minimum allowed graph size.
```

Comment:
Is this enabled by default? Or requires some parameter / option?

Reply:
To my understanding, it does not need additional configuration.